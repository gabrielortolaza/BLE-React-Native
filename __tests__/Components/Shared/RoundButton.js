import React from "react";
import { Platform } from "react-native";
import renderer from "react-test-renderer";

const RoundButton = require("../../../src/Components/Shared/RoundButton").default;

describe("RoundButton", () => {
  it("Default", () => {
    expect(renderer.create(<RoundButton onPress={() => {}}>{ "Test " + Platform.OS}</RoundButton>).toJSON()).toMatchSnapshot();
  });
/*
  it('Wide', () => {
    expect(renderer.create(<RoundButton wide onPress={() => {}}>{ 'Test ' + Platform.OS }</RoundButton>).toJSON()).toMatchSnapshot()
  })

  it('Small', () => {
    expect(renderer.create(<RoundButton small onPress={() => {}}>{ 'Test ' + Platform.OS }</RoundButton>).toJSON()).toMatchSnapshot()
  })

  it('White', () => {
    expect(renderer.create(<RoundButton white onPress={() => {}}>{ 'Test ' + Platform.OS }</RoundButton>).toJSON()).toMatchSnapshot()
  })

  it('Facebook', () => {
    expect(renderer.create(<RoundButton facebook onPress={() => {}}>{ 'Test ' + Platform.OS }</RoundButton>).toJSON()).toMatchSnapshot()
  })

  it('Minimal', () => {
    expect(renderer.create(<RoundButton minimal onPress={() => {}}>{ 'Test ' + Platform.OS }</RoundButton>).toJSON()).toMatchSnapshot()
  })

  it('Light Blue', () => {
    expect(renderer.create(<RoundButton lightBlue onPress={() => {}}>{ 'Test ' + Platform.OS }</RoundButton>).toJSON()).toMatchSnapshot()
  })

  it('Transparent', () => {
    expect(renderer.create(<RoundButton transparentBackground onPress={() => {}}>{ 'Test ' + Platform.OS }</RoundButton>).toJSON()).toMatchSnapshot()
  })

  it('Shadow', () => {
    expect(renderer.create(<RoundButton shadow onPress={() => {}}>{ 'Test ' + Platform.OS }</RoundButton>).toJSON()).toMatchSnapshot()
  })

  it('Warm Grey Text', () => {
    expect(renderer.create(<RoundButton warmGreyText onPress={() => {}}>{ 'Test ' + Platform.OS }</RoundButton>).toJSON()).toMatchSnapshot()
  })

  it('Blue Text', () => {
    expect(renderer.create(<RoundButton blueText onPress={() => {}}>{ 'Test ' + Platform.OS }</RoundButton>).toJSON()).toMatchSnapshot()
  })

  it('OVerride', () => {
    expect(renderer.create(<RoundButton override onPress={() => {}}><Text>{ 'Test ' + Platform.OS }</Text></RoundButton>).toJSON()).toMatchSnapshot()
  })

  it('Button Style', () => {
    expect(renderer.create(<RoundButton buttonStyle={{width: 300}} onPress={() => {}}>{ 'Test ' + Platform.OS }</RoundButton>).toJSON()).toMatchSnapshot()
  })

  it('Text Style', () => {
    expect(renderer.create(<RoundButton textStyle={{fontSize: 30}} onPress={() => {}}>{ 'Test ' + Platform.OS }</RoundButton>).toJSON()).toMatchSnapshot()
  })
  */
});
