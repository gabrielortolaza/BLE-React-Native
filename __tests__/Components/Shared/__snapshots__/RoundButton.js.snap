// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundButton Default 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "rgb(66, 162, 194)",
          "borderRadius": 44,
        },
      ],
      Object {
        "overflow": null,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "rgb(66, 162, 194)",
        "borderRadius": 44,
        "opacity": 1,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "justifyContent": "center",
          },
          Object {
            "flex": 0,
            "height": 88,
            "minWidth": 212,
            "paddingHorizontal": 44,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Array [
              Object {
                "color": "rgb(255, 255, 255)",
                "fontFamily": "Graphik-Semibold",
                "fontSize": 36,
                "height": 88,
                "lineHeight": 88,
                "marginBottom": 0,
                "textAlign": "center",
                "textAlignVertical": "center",
              },
            ],
          ]
        }
      >
        Test ios
      </Text>
    </View>
  </View>
</View>
`;
